lua/lspconfig/configs.lua

```diff
@@ -98,11 +98,17 @@ function configs.__newindex(t, config_name, config_def)
 
     if config.autostart == true then
       local event_conf = config.filetypes and { event = 'FileType', pattern = config.filetypes }
-        or { event = 'BufReadPost' }
+        or { event = { 'BufReadPost', 'BufNewFile' } }
       api.nvim_create_autocmd(event_conf.event, {
         pattern = event_conf.pattern or '*',
         callback = function(opt)
-          M.manager:try_add(opt.buf)
+          -- Use vim.schedule() to ensure filetype detection happens first.
+          -- Sometimes, BufNewFile triggers before 'filetype' is set.
+          vim.schedule(function()
+            if api.nvim_buf_is_valid(opt.buf) then
+              M.manager:try_add(opt.buf)
```

Comment:
shouldn't try_add / try_add_wrapper  check `nvim_buf_is_valid`? they are "try" patterns, after all.

Reply:
I considered this myself, actually, but didn't have a strong opinion either way.

- Both functions would need a check, so code size/complexity is about the same either way.
- Moving the checks might prevent another bug like this in the future.
- `try_add_wrapper()` would do the check twice, but that's not really a problem.

If your intuition is to move the checks, I might as well.